mypy/checker.py

```diff
@@ -2482,9 +2482,37 @@ def check_assignment_to_multiple_lvalues(self, lvalues: List[Lvalue], rvalue: Ex
             # using the type of rhs, because this allowed more fine grained
             # control in cases like: a, b = [int, str] where rhs would get
             # type List[object]
-
-            rvalues = rvalue.items
-
+            rvalues = []  # type: List[Expression]
+            lhs_len = len(lvalues)
+            idx_of_iterable = []
+            item_type_of_iterable = []  # type: List['mypy.types.Type']
+            idx = 0
+            for rval in rvalue.items:
+                if isinstance(rval, StarExpr):
+                    typs = get_proper_type(self.expr_checker.visit_star_expr(rval).type)
+                    if isinstance(typs, TupleType):
+                        rvalues.extend([TempNode(typ) for typ in typs.items])
+                        lhs_len -= len(typs.items)
+                        idx += len(typs.items)
+                    elif self.type_is_iterable(typs) and isinstance(typs, Instance):
+                        item_type_of_iterable.append(self.iterable_item_type(typs))
+                        idx_of_iterable.append(idx)
+                    else:
+                        self.fail("StarExpr should not be a '{}'".format(typs), context)
```

Comment:
Nits: StarExpr is an internal name that we'd rather not use in an error message. It would be better to use something more descriptive, such as `Invalid type 'x' for *expr (iterable expected)`

If you pass `rval` as the context to `fail`, the error message should point to the relevant subexpression, making it easier to figure out what it going on.

It would be good to have a test case for this as well.